trieste/acquisition/utils.py

```diff
@@ -48,7 +48,8 @@ def wrapper(x: TensorType) -> TensorType:
         if length == 0:
             return fn(x)
 
-        elements_per_block = tf.size(x) / length
+        # Use int64 to calculate the input tensor size, otherwise we can overflow for large tensors.
```

Comment:
this is good but could you please add in the docstrings few sentences about memory requirements wrt scaling?

Reply:
Do you mean docstring for `split_acquisition_function`? Could you give an example please? I could say that the memory usage of tensor `x` is its flattened size times 8, but that is fairly obvious.
Or do you mean in the optimizer docstring?

Reply:
i think it would be more appropriate in the `generate_continuous_optimizer` docstring where tiled_candidates becomes very big - I was thinking of an additional note with a warning that using a large num_initial_samples with high dimensional search space and/or batch sizes will require a lot of memory.

Reply:
I have added a note.